i3pystatus/network.py

```diff
@@ -410,31 +417,39 @@ def run(self):
             format_values['kbs'] = "{0:.1f}".format(round(kbs, 2))
 
-            if self.separate_color and self.pango_enabled:
-                color = self.color_up
-                color_template = "<span color=\"{}\">{}</span>"
-                per_recv = network_usage["bytes_recv"] * self.divisor / (self.recv_limit * 1024)
-                per_sent = network_usage["bytes_sent"] * self.divisor / (self.sent_limit * 1024)
-                c_recv = self.get_gradient(int(per_recv * 100), self.colors, 100)
-                c_sent = self.get_gradient(int(per_sent * 100), self.colors, 100)
-                format_values["bytes_recv"] = color_template.format(c_recv, network_usage["bytes_recv"])
-                format_values["bytes_sent"] = color_template.format(c_sent, network_usage["bytes_sent"])
-                if self.graph_type == 'output':
-                    c_kbs = c_sent
+            if self.dynamic_color:
+                if self.separate_color and self.pango_enabled:
+                    color = self.color_up
+                    color_template = "<span color=\"{}\">{}</span>"
+                    per_recv = network_usage["bytes_recv"] * self.divisor / (self.recv_limit * 1024)
+                    per_sent = network_usage["bytes_sent"] * self.divisor / (self.sent_limit * 1024)
+                    c_recv = self.get_gradient(int(per_recv * 100), self.colors, 100)
+                    c_sent = self.get_gradient(int(per_sent * 100), self.colors, 100)
+                    format_values["bytes_recv"] = color_template.format(c_recv, network_usage["bytes_recv"])
+                    format_values["bytes_sent"] = color_template.format(c_sent, network_usage["bytes_sent"])
+                    if self.graph_type == 'output':
+                        c_kbs = c_sent
+                    else:
+                        c_kbs = c_recv
+                    format_values['network_graph'] = color_template.format(c_kbs, format_values["network_graph"])
+                    format_values['kbs'] = color_template.format(c_kbs, format_values["kbs"])
                 else:
-                    c_kbs = c_recv
-                format_values['network_graph'] = color_template.format(c_kbs, format_values["network_graph"])
-                format_values['kbs'] = color_template.format(c_kbs, format_values["kbs"])
+                    percent = int(kbs * 100 / limit)
+                    color = self.get_gradient(percent, self.colors, 100)
             else:
-                percent = int(kbs * 100 / limit)
-                color = self.get_gradient(percent, self.colors, 100)
+                color = None
         else:
             color = None
 
         if sysfs_interface_up(self.interface, self.unknown_up):
             if not color:
                 color = self.color_up
             format_str = self.format_up
+
+            if self.detect_active:
+                for pattern in self.format_active_up:
```

Comment:
Do you think it makes sense to sort the keys and have a break if there is a match? That way the order of evaluation is always the same and it will always set the format_string to the first match.